### ISSUE TITLE
Doc: AdamW docstring is different from documentation on keras and tensorflow website

### ISSUE THREAD

**Issue by AjaniStewart** [keras-team/keras#18712]
From [Keras](https://keras.io/api/optimizers/adamw/) documentation on AdamW:

> weight_decay: Float, defaults to None. If set, weight decay is applied.

However, the AdamW constructor raises a ValueError if `weight_decay` is None

```
if self.weight_decay is None:
            raise ValueError(
                "Missing value of `weight_decay` which is required and"
                " must be a float value.")
```

**Comment by SuryanarayanaY** [comment]
Hi @AjaniStewart ,

It seems documentation bug. It is copying the arguments from the Parent class arguments. Will look into it. Thanks!
